src/libultra/flash/osFlash.c

```diff
@@ -1,41 +1,37 @@
 #include "prevent_bss_reordering.h"
 #include "ultra64.h"
 #include "global.h"
+#include "PR/os_internal_flash.h"
 
-s32 framDeviceInfo[4];
-OSIoMesg framDeviceInfoQuery;
+u32 __osFlashID[4];
+OSIoMesg __osFlashMsg;
 OSMesgQueue __osFlashMessageQ;
 OSPiHandle __osFlashHandler;
-OSMesg __osFlashMsgBuf;
+OSMesg __osFlashMsgBuf[1];
 s32 __osFlashVersion;
-UNK_TYPE1 D_801FD0FC[0x14];
-
-typedef enum {
-    /* 0 */ FLASH_OLD,
-    /* 1 */ FLASH_NEW
-} FlashVersion;
+static s32 sBssPad[5];
 
 u32 osFlashGetAddr(u32 pageNum) {
-    s32 addr = (__osFlashVersion == FLASH_OLD) ? pageNum << 6 : pageNum << 7;
+    u32 addr = (__osFlashVersion == OLD_FLASH) ? pageNum * 64 : pageNum * 128;
```

Comment:
Should this also incorporate `FLASH_BLOCK_SIZE` somehow?

Reply:
Hmm, I think that makes sense but not 100% sure. Thoughts @Thar0

Reply:
The discrepancy between old and new flash here is I believe due to a hardware bug in the old flash where the address bits are shifted 1-off from where they should actually be. I think using `FLASH_BLOCK_SIZE >> 1` and `FLASH_BLOCK_SIZE` and a brief comment makes sense.